_rules/text-node-contrast-afw4f7.md

```diff
@@ -0,0 +1,337 @@
+
+**Note**: Passing this rule does not mean that the [text node](https://dom.spec.whatwg.org/#text) has sufficient color contrast. White text on a black and white image passes this rule, whether or not that text is legible depends on the how much of the white words are positioned on the black parts of the image. This requires further testing.
+
+## Assumptions
+
+Success criterion 1.4.3 has exceptions for "incidental" text, which includes inactive user interface components and decorative texts. The rule assumes that [text nodes](https://dom.spec.whatwg.org/#text) should be ignored are hidden from assistive technologies. If this isn't the case, the rule may produce incorrect results.
+
+Success criterion 1.4.3 also has an exception for logos and brand names. Since logos and brand names are usually displayed through images to ensure correct rendering, this rule does not take logos or brand names into consideration. If a logo or brand name is included using [text nodes](https://dom.spec.whatwg.org/#text), this rule may not produce the rule may produce incorrect results.
+
+## Accessibility Support
+
+Different browsers have different levels of support for CSS. This can cause contrast issues in one browser that do not appear in another. Because of that, this rule can produce different results depending on the browser that is used.
+
+## Background
+
+- [Understanding Success Criterion 1.4.3: Contrast (Minimum)](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html)
+- [G18: Ensuring that a contrast ratio of at least 4.5:1 exists between text (and images of text) and background behind the text](https://www.w3.org/WAI/WCAG21/Techniques/general/G18)
+- [G145: Ensuring that a contrast ratio of at least 3:1 exists between text (and images of text) and background behind the text](https://www.w3.org/WAI/WCAG21/Techniques/general/G145)
+- [F83: Failure of Success Criterion 1.4.3 and 1.4.6 due to using background images that do not provide sufficient contrast with foreground text (or images of text)](https://www.w3.org/WAI/WCAG21/Techniques/failures/F83)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+Black [text node](https://dom.spec.whatwg.org/#text) on a white background.
+
+```html
+<p style="color: #333; background: #FFF;">
+	Some text in English
+</p>
+```
+
+#### Passed Example 2
+
+Black [text node](https://dom.spec.whatwg.org/#text) on a partially white gradient background.
+
+```html
+<p style="color: #333; background: linear-gradient(to right, #FFF, #00F);">
+	Some text in English
+</p>
+```
+
+#### Passed Example 3
+
+Light [text node](https://dom.spec.whatwg.org/#text) on a partially black background image.
+
+```html
+<p
+	style="color: #CCC; height:50px; padding-top:25px; background: #000 no-repeat -20px -20px url('../test-assets/black-hole.jpeg');"
```

Comment:
can we add the test-assets under a specific directory pertaining to the ruleId?

eg: `test-assets/XXXXXX/something.xyz`